discovery_engine_core/core/src/stack/exploration.rs

```diff
@@ -12,504 +12,150 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-use std::collections::{BTreeSet, HashSet};
-
-use displaydoc::Display;
-use itertools::{chain, Itertools};
-use ndarray::{Array2, ArrayView1};
-use rand::{prelude::IteratorRandom, Rng};
-use thiserror::Error;
-use xayn_ai::{
-    cosine_similarity,
-    ranker::{pairwise_cosine_similarity, CoiPoint, NegativeCoi, PositiveCoi},
+//! Exploration stack.
```

Comment:
I would prefer to keep it out of `ops` because most of the code there is to fetch and filter new articles and that part is missing from the exploration stack.